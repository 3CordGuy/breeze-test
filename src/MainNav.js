import React from "react";
import { Link } from "@reach/router";

const MainNav = (props) => {
    console.log(props);

    /* 
      I realize this is not the proper api for matching routes with Reach router
      but I need to move on to other important features to complete this challenge
    */
    const get_item_class_for_route = (route) => {
        return props["*"].split("/")[0] === route ? "active item" : "item";
    };

    return (
        <div className="ui attached tabular menu">
            <Link to="/groups" className={get_item_class_for_route("groups")}>
                Groups
            </Link>
            <Link to="/people" className={get_item_class_for_route("people")}>
                People
            </Link>
        </div>
    );
};

export default MainNav;
