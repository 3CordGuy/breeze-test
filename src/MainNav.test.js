import React from "react";
import { shallow } from "enzyme";
import MainNav from "./MainNav";

let wrapper;

describe("<MainNav />", () => {
    beforeAll(() => {
        wrapper = shallow(<MainNav />);

        wrapper.setProps({ "*": "" });
    });

    test("should match the snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
