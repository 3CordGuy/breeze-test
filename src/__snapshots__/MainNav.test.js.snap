// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNav /> should match the snapshot 1`] = `
<div
  className="ui attached tabular menu"
>
  <Link
    className="active item"
    to="/"
  >
    All Groups
  </Link>
  <Link
    className="item"
    to="/people"
  >
    All People
  </Link>
</div>
`;
